sln, vcxproj, props and targets for Rider to work on all OS

van800 opened the pull request:
Automatically resolve C++ standard libraries.
I have tested on Mac and Windows.
Just need to Build once to have `gen` code appear.

van800 commented:
I see, what you mean. Thank you.
It is intended for developers creating an extension that uses godot-cpp
`NMakePreprocessorDefinitions` here is copy-paste from the engine. I will drop it.
I will make other changes too, completely forgot about scenarios like compiling gdextension for a different platform and arch.

dsnopek commented:
I wonder if this might be better as a community-maintained template for users of Rider?

I don't know if anyone on the GDExtension team uses Rider, and would be able to test PRs for it or maintain it going foward. Whereas a small community of enthusiastic users of Rider could take that on

van800 commented:
It is up to you to decide if you'd like to incorporate the proposed changes.

https://github.com/user-attachments/assets/12bbf6e1-3a9c-45be-8350-98b685a8ceaf

A bit of me struggling with typing code. But the experience of GDExtension and the game in one place might be very good after all.

van800 commented:
Added a fix for debugging to work.
<img width="2560" height="931" alt="image" src="https://github.com/user-attachments/assets/5222d1e5-62a2-43c8-b6a6-5f7008a98a39" />

dsnopek commented:
Discussed at the GDExtension team meeting, and we're a little wary of adding so much code for Rider support. We'd prefer either:

1. Adding something to SCons (in godot-cpp) to generate these files (similar to `scons vsproj=true` in the Godot project) or a shell script or even docs for how to generate these files. Or,
2. Adding a page to the documentation for developers about how to set this up for themselves (similar to [these docs](https://docs.godotengine.org/en/latest/contributing/development/configuring_an_ide/rider.html) for the Godot project). Or,
3. Adding the minimum files here to make this work (even if it has less functionality)

van800 commented:
@dsnopek I went with option (3).
All files needed for Rider are now in the `.sln` folder, which is configured to be invisible in vscode.
Would this solution fit?

Just a side note. It forced me to do some changes in the Rider Godot integration, so for Rider `demo` project would be fully recognized by Rider 2025.3, not the current one 2025.2.

van800 commented:
I have removed some auxilary files and macos_debug info, added a readme, describing each of the files in the `.sln` folder. 
Does it help to justify the change? @dsnopek 

I am afraid, there is nothing else to remove or simplify. I can merge targets and props into gdext.vcxproj, but it would not simplify understanding how msbuild files work.

van800 commented:
Reworked sln-vcxproj mapping again and it became a lot simpler, I am surprised myself.

@paddy-exe 
The .sln file is created when a new solution is created in VS or Rider and updated with editing solution configurations in VS or Rider. The corresponding part of the .vcxproj, which maps $(Configuration) and $(Platform) to other properties, is typically generated by IDE, though it can be customized manually if needed.

In the video I wanted to point out how switching solution configuration affects the resulting scons call. And then I saw leftovers of previous refactoring and removed them.
https://youtu.be/zC5Cy1BZp0Y

CadeEvs commented:
Was testing this PR out with 2025.3 EAP4 and running into a couple issues,

1. Trying to build the sln results in a compile error
```
0>------- Started building project: gdext
scons platform=windows arch=x86_64 target=template_debug dev_build=yes
'scons' is not recognized as an internal or external command,
operable program or batch file.
0>Microsoft.MakeFile.Targets(45,5): Error MSB3073 : The command "scons platform=windows arch=x86_64 target=template_debug dev_build=yes" exited with code 9009.
0>------- Finished building project: gdext. Succeeded: False. Errors: 1. Warnings: 0
```
2. In `example_class.h` I'm getting an error at line 10 for GDCLASS
```
Cannot resolve symbol 'uint32_t'
```
3. Rider doesn't seem to be helping autocomplete/add the header files for Godot classes



https://github.com/user-attachments/assets/b60dd949-b7d0-4f54-8088-9419030f8391
